security/tasks.py

```diff
@@ -29,99 +31,257 @@ class LoggedTask(Task):
 
     abstract = True
     logger_level = logging.WARNING
+    retry_error_message = (
+        'Task "{task_name}" ({task}) failed on exception: "{exception}", attempt: "{attempt}" and will be retried'
+    )
+    fail_error_message = 'Task "{task_name}" ({task}) failed on exception: "{exception}", attempt: "{attempt}"'
+    stale_time_limit = None
+    # Support set retry delay in list. Retry countdown value is get from list where index is attempt
+    # number (request.retries)
+    default_retry_delays = None
 
-    def push_request(self, *args, **kwargs):
-        task_id = self.request.id
-        output_stream = self.request.output_stream
-        super().push_request(*args, **kwargs)
-        self.request.id = task_id
-        self.request.output_stream = output_stream
-
-    def get_task(self):
-        return CeleryTaskLog.objects.get(pk=self.request.id)
+    @property
+    def task_log(self):
+        return CeleryTaskLog.objects.get(
+            celery_task_id=str(self.request.id),
+            retries=self.request.retries
+        )
 
     def __call__(self, *args, **kwargs):
-        # Every set attr is send here
+        """
+        Overrides parent which works with thread stack. We didn't want to allow change context which was generated in
+        one of apply methods. Call task directly is now disallowed.
+        """
+        req = self.request_stack.top
+
+        if not req or req.called_directly:
+            raise CeleryError(
+                'Task cannot be called directly. Please use apply, apply_async or apply_async_on_commit methods'
+            )
```

Comment:
I disallow some type of celery usage. It was nonsence to call celery task function directly